agent/command/s3_put.go

```diff
@@ -464,3 +491,26 @@ func (s3pc *s3put) isPublic() bool {
 	return (s3pc.Visibility == "" || s3pc.Visibility == artifact.Public) &&
 		(s3pc.Permissions == s3.BucketCannedACLPublicRead || s3pc.Permissions == s3.BucketCannedACLPublicReadWrite)
 }
+
+func (s3pc *s3put) fileExists(remoteName string) (bool, error) {
```

Comment:
I would call this something like `remoteFileExists` so that it doesn't cause ambiguity between the local files vs. remote files.